File: GFG/050125/JS/CountPairWithSumLessThanTarget.js
// Problem Link: https://www.geeksforgeeks.org/problems/count-pairs-whose-sum-is-less-than-target/1

class Solution {
    countPairs(arr, target) {
        // write your code here
        arr.sort((a, b)=> a-b);  //   custom comparator
       
        let l=0;
        let r= arr.length-1;
        let cnt=0;
        
        while(l<r){
            
            if(arr[l]+arr[r] < target){
                cnt+= (r-l);
                l++;
            }else{
                r--;
            }
        }
        
        return cnt;
    }
}

File: GFG/050125/Python/CountPairWithSumLessThanTarget.py
// Problem Link: https://www.geeksforgeeks.org/problems/count-pairs-whose-sum-is-less-than-target/1

class Solution:
    #Complete the below function
    def countPairs(self, arr, target):
        #Your code here
        arr.sort()
        l=0
        r= len(arr)-1
        cnt=0
        while(l<r):
            if(arr[l]+arr[r] < target):
                cnt+= (r-l)
                l+=1
            else:
               r-=1
        return cnt
